lms/djangoapps/email_marketing/tasks.py

```diff
@@ -0,0 +1,140 @@
+"""
+This file contains celery tasks for email marketing signal handler.
+"""
+import logging
+import datetime
+import time
+
+from pytz import UTC
+
+from celery import task
+from django.contrib.auth.models import User
+
+from student.models import UserProfile
+from email_marketing.models import EmailMarketingConfiguration
+
+from sailthru.sailthru_client import SailthruClient
+from sailthru.sailthru_error import SailthruClientError
+
+log = logging.getLogger(__name__)
+
+
+# pylint: disable=not-callable
+@task(bind=True, default_retry_delay=3600, max_retries=24)
+def update_user(self, username, new_user=False):
+    """
+    Adds/updates Sailthru profile information for a user.
+     Args:
+        username(str): A string representation of user identifier
+    Returns:
+        None
+    """
+    email_config = EmailMarketingConfiguration.current()
+    if not email_config.sailthru_enabled:
+        return
+
+    # get user
+    user = User.objects.get(username=username)
+    if not user:
+        log.error("User not found duing Sailthru update %s", username)
+        return
+
+    # ignore anonymous users
+    if user.is_anonymous():
+        return
+
+    # get profile
+    profile = UserProfile.objects.get(user=user)
```

Comment:
You can actually get this via `user.profile` because it's connected with a `OneToOneField`. Also, if you load the user using `User.objects.select_related('profile')`, then django will only hit the database once to populate both objects.